sw/airborne/subsystems/imu/imu_px4fmu_v2.4.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2013-2016 the paparazzi team
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+
+/**
+ * @file subsystems/imu/imu_px4fmu_v2.4.h
+ * Driver for pixhawk IMU's.
+ * L3GD20H + LSM303D (both on spi)
+ */
+
+#include "subsystems/imu.h"
+#include "subsystems/abi.h"
+#include "mcu_periph/spi.h"
+#include "peripherals/l3gd20_regs.h"
+#include "peripherals/lsm303dlhc_regs.h"
+#include "peripherals/lsm303dlhc_spi.h"
+
+/* SPI defaults set in subsystem makefile, can be configured from airframe file */
+PRINT_CONFIG_VAR(IMU_LSM_SPI_SLAVE_IDX)
+PRINT_CONFIG_VAR(IMU_L3G_SPI_SLAVE_IDX)
+PRINT_CONFIG_VAR(IMU_SPI_DEV)
+
+struct ImuPX4 imu_px4;
+
+void imu_impl_init(void)
+{
+
+  /* L3GD20 gyro init */
+  /* initialize gyro and set default options */
+  l3gd20_spi_init(&imu_px4.l3g, &IMU_SPI_DEV, IMU_L3G_SPI_SLAVE_IDX);
+
+  /* LSM303dlhc acc + magneto init */
+  lsm303dlhc_spi_init(&imu_px4.lsm_acc, &IMU_SPI_DEV, IMU_LSM_SPI_SLAVE_IDX, LSM_TARGET_ACC);
+#if !MODULE_HMC58XX_UPDATE_AHRS
```

Comment:
We clearly need better/proper support for multiple IMUs and get rid of the global `imu` struct to really solve this problem...

While I can understand the desire to keep the amount of configuration low, I personally prefer something more generic so that you can also disable the px4 mag if you use a different external GPS.

One way to achieve that could be to add a define/flag `IMU_PX4_DISABLE_MAG` and check for that.
Then if it is not defined, but `MODULE_HMC58XX_UPDATE_AHRS` is defined to `TRUE`, you can set `IMU_PX4_DISABLE_MAG` to true as well...
That way it would still "automatically" work with the hmc module, but give the option to explicitly disable the px4 mag even if you use a different external one...